frontend/src/pages/register.jsx

```diff
@@ -19,126 +18,115 @@ const Heading = () => (
-        className="mt-2 px-3 py-2 w-full border rounded-lg focus:outline-none focus:ring-2 focus:ring-blue-500"
-      />
-      {errors.username && <p className="text-red-500 text-sm">{errors.username.message}</p>}
-    </div>
-
-    <div>
-      <label htmlFor="email" className="block text-lg font-medium text-gray-700">Email</label>
-      <input
-        id="email"
-        type="email"
-        {...register('email', { required: 'Email is required' })}
-        className="mt-2 px-3 py-2 w-full border rounded-lg focus:outline-none focus:ring-2 focus:ring-blue-500"
-      />
-      {errors.email && <p className="text-red-500 text-sm">{errors.email.message}</p>}
-    </div>
-
-    <div>
-      <label htmlFor="phone" className="block text-lg font-medium text-gray-700">Phone Number</label>
-      <input
-        id="phone"
-        type="tel"
-        {...register('phone', {
-          required: 'Phone number is required',
-          pattern: { value: /^[0-9]{10}$/, message: 'Invalid phone number' },
-        })}
-        className="mt-2 px-3 py-2 w-full border rounded-lg focus:outline-none focus:ring-2 focus:ring-blue-500"
-      />
-      {errors.phone && <p className="text-red-500 text-sm">{errors.phone.message}</p>}
-    </div>
-
-    <div>
-      <label htmlFor="address" className="block text-lg font-medium text-gray-700">Address</label>
-      <input
-        id="address"
-        type="text"
-        {...register('address', { required: 'Address is required' })}
-        className="mt-2 px-3 py-2 w-full border rounded-lg focus:outline-none focus:ring-2 focus:ring-blue-500"
-      />
-      {errors.address && <p className="text-red-500 text-sm">{errors.address.message}</p>}
-    </div>
-
-    <div>
-      <label htmlFor="password" className="block text-lg font-medium text-gray-700">Password</label>
-      <input
-        id="password"
-        type="password"
-        {...register('password', { required: 'Password is required' })}
-        className="mt-2 px-3 py-2 w-full border rounded-lg focus:outline-none focus:ring-2 focus:ring-blue-500"
-      />
-      {errors.password && <p className="text-red-500 text-sm">{errors.password.message}</p>}
-    </div>
-
-    <button
-      type="submit"
-      className="w-full px-6 py-3 bg-blue-600 text-white font-bold rounded-lg hover:bg-blue-700 transition duration-300 ease-in-out"
-      disabled={isLoading}
-    >
-      {isLoading ? 'Creating account...' : 'Create Account'}
-    </button>
-  </form>
-);
-
-// Register: The main registration page
 const Register = () => {
   const { register, handleSubmit, reset, formState: { errors } } = useForm();
-  const [registerUser, { isLoading }] = useRegisterUserMutation();
+  const [registerUser, { isLoading }] = useRegisterUserMutation(); // Updated hook usage
   const navigate = useNavigate();
 
-  // Use useCallback to memoize the onSubmit handler
-  const onSubmitHandler = React.useCallback(async (data) => {
+  const onSubmit = async (data) => {
```

Comment:
Not sure whether this data is available here. you have passed as an argument rather than destructuring